### ISSUE TITLE
False positive IDE0047 : "Parentheses can be removed" in nested expression of interpolated string

### ISSUE THREAD

**Issue by WolfgangKluge** [dotnet/roslyn#33038]
**Version Used**:  Visual Studio 2017 (15.8.9)

**Steps to Reproduce**:
Analyze the following code
```c#
var nl = System.Environment.NewLine;
var z = $@"
    {(string.IsNullOrEmpty(nl)
        ? ""
        : $"{(string.IsNullOrEmpty(nl) ? 1 : 2)}"
    )}
";
```
Looks like
<img src="https://user-images.githubusercontent.com/639482/52115380-c8b1bd80-260e-11e9-98ec-59cc6591a3f7.png" width="400" />

Despite the uselessness of the code above, the inner interpolated expression should not be catched by the analyzer and marked with IDE0047. If one removes the parentheses, the compiler finds several errors.

The outer expression needs parentheses in order to get this false positive.

**Expected Behavior**:
No info/warning about superflous parentheses should be visible

**Actual Behavior**:
The inner expression is marked with IDE0047

**Comment by WolfgangKluge** [comment]
In addition, if I try to suppress those messages in the source (`#pragma`) with the quick actions-tool, then the code is faulty, too..

<img src="https://user-images.githubusercontent.com/639482/52116096-9bfea580-2610-11e9-9283-cde8ecb53e09.png" width="400" />

**Comment by CyrusNajmabadi** [comment]
@sharwell Does this repro for you?  it does for me in the latest roslyn bits.

![image](https://user-images.githubusercontent.com/4564579/52146742-8772db00-2619-11e9-953f-11d446cf9492.png)

Note: if i add additional parens, then i correctly get:

![image](https://user-images.githubusercontent.com/4564579/52146766-9bb6d800-2619-11e9-8814-b745a2077108.png)

However, even doing 'fix all' is smart enough to know that even though both of those parens are marked as unnecessary, only one set can be removed to not break anything.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/32284
